### PR TITLE
Move enclave CLI logic into ev-cli crate

### PR DESCRIPTION
**Pull request by doylemark**
# Why
* Move CLI commands out of `ev-enclave` into `ev-cli` crate
* Resolves ETR-1973
# How
* Move the generic api client and it's CAPI/PAPI impl into `common`.
* Move all the `ev-enclave/cli` mod commands into `ev-cli`
* Move stuff around so it all works together

#### Note
Only commits after #fc57a02 are relevant to this pr, branch is based off #106

### CODE REVIEW

File: crates/ev-cli/Cargo.toml [hunk 1]
```diff
@@ -16,4 +16,11 @@ human-panic = "1.0.3"
 log = "0.4.17"
 env_logger = "0.9.0"
 exitcode = "1.1.2"
+dialoguer = "0.10.2"
+attestation-doc-validation = "0.7.4"
```
Comment (doylemark) [inline crates/ev-cli/Cargo.toml:20]: Would be nice to remove some deps like this, and move all EnclaveApi stuff into `ev-enclave`, but it's outside scope of this PR. Objective here is just to do what's necessary to move the cli commands themselves into `ev-cli`

File: crates/common/src/api/client.rs [hunk 2]
```diff
@@ -184,6 +183,19 @@ impl std::fmt::Display for ApiErrorKind {
     }
 }
 
+impl crate::CliError for ApiError {
+    fn exitcode(&self) -> exitcode::ExitCode {
```
Comment (lfarrel6) [inline crates/common/src/api/client.rs:187]: Not needed in this PR, but can we remove the `exitcode` dependency? Can be vendored, it's just a set of constants.

File: crates/ev-enclave/src/version/mod.rs [hunk 3]
```diff
@@ -41,51 +39,6 @@ pub fn get_latest_major_version() -> Result<u8, VersionError> {
     Ok(env!("CARGO_PKG_VERSION_MAJOR").parse::<u8>()?)
```
Comment (lfarrel6) [inline crates/ev-enclave/src/version/mod.rs:39]: does this return the enclave major or the CLI major?

**Review by lfarrel6 — APPROVED** [review]

File: crates/common/src/api/client.rs (hunk 2 above)
Comment (doylemark) [reply crates/common/src/api/client.rs:187]: yup 🔥

File: crates/ev-enclave/src/version/mod.rs (hunk 3 above)
Comment (doylemark) [reply crates/ev-enclave/src/version/mod.rs:39]: @lfarrel6 CLI Major, need to figure out how we're gonna deal with ev-enclave <-> ev-cli versioning. Imo we need to disassociate the CLI cargo version from the behaviour of the enclave cli.

**Pull request merged**